index.js

```diff
@@ -0,0 +1,160 @@
+'use strict'
+
+const _ = require('lodash')
+const uuid = require('uuid')
+
+class Senatus {
+  constructor (opts) {
+    this._storage = opts.storage
+    this._notifier = opts.notifier
+    this._sigAlgo = opts.sigAlgo
+    this._whitelistKey = opts.whitelistKey
+
+    this._whitelist = this._buildWhitelistMap()
+  }
+
+  _buildWhitelistMap () {
```

Comment:
this function is mixing sync and async, and is called from the constructor.

also it throws at some points during async operations and loops, which would lead to a server crash. not sure if an invalid remotely stored whitelist should lead to a server crash. 

right now it also seems that you need to intantiate a new instance in case you want a new whitelist

Reply:
good points. I've updated accordingly. Since whitelist is stored as mutable item, the whitelist itself can be updated without creating another senatus instance.